Sources/docc/DocCDocumentation.docc/documenting-api-with-different-language-representations.md

```diff
@@ -52,12 +52,43 @@ func doSomething(with someValue: Int) throws
+DocC automatically adds a generic description for the "error" parameter for the Objective-C version of that symbol's page. 
 
-You don't need to document the Objective-C representation's "error" parameter or Objective-C specific return value for symbols defined in Swift.
-DocC automatically adds a generic description for the "error" parameter and extends your return value documentation to describe the Objective-C specific return value behavior. 
+When the Swift representation returns `Void`---which corresponds to a `BOOL` return value in Objective-C (like in the example above)---DocC adds a generic description of the `BOOL` return value to the Objective-C version of that symbol's page. 
 
-If you want to customize this documentation you can manually document the "error" parameter and return value. 
+When the Swift representation returns a value---which corresponds to a `nullable` return value in Objective-C---DocC extends your return value documentation, for the Objective-C version of that symbol's page, to describe that the methods returns `nil` if an error occurs unless your documentation already covers this behavior. 
+For example, consider this throwing function in Swift and its corresponding Objective-C interface:
+
+**Swift definition**
+
+```swift
+/// - Parameters:
+///   - someValue: Some description of this parameter.
+/// - Returns: Some description of this return value.
+@objc func doSomething(with someValue: Int) throws -> String
+```
+
+**Generated Objective-C interface**
+
+```objc
+- (nullable NSString *)doSomethingWith:(NSInteger)someValue
+                                 error:(NSError **)error;
+```
+
+For the Swift representation, with one parameter and a non-optional return value,
+DocC displays the "someValue" parameter documentation and return value documentation as-is.
+For the Objective-C representation, with two parameters and a nullable return value,
+DocC displays the "someValue" parameter documentation, generic "error" parameter documentation, and extends the return value documentation with a generic description about the Objective-C specific `nil` return value when the method encounters an error. 
+
+If you want to customize this documentation you can manually document the "error" parameter and return value.
 Doing so won't change the Swift version of that symbol's page.
 DocC will still hide the parameter and return value documentation that doesn't apply to each source language's version of that symbol's page.
 
+If your return value documentation already describes the `nil` return value, DocC won't extent it with the generic `nil` value description.
+
+> Note:
+> If you document the Objective-C specific `nil` return value for a method that correspond to a throwing function in Swift, 
+> DocC will display that documentation as-is on the Swift version of that page where is has a non-optional return type.
```

Comment:
The cases read a bit confusing to me, because they're explained in different parts of the article -- am I understanding it right that if I manually document the return value, it **won't** show in the Swift documentation unless it's documentation for a throwing function? 

Is this behaviour intentional? Should we also hide the extra line for throwing functions?

Reply:
No, if the developers documents the Objective-C specific `nil` return value, it _will_ display for both Swift and Objective-C. 

This aside is meant to highlight that unless the authored documentation is a big improvement over the generic phrase that DocC adds, developers should probably avoid documenting Objective-C specific `nil` return values.

Reply:
Maybe we could mention that explicitly here?

Reply:
I rephrased this documentation in https://github.com/swiftlang/swift-docc/pull/1096/commits/573f1c60cd36d4619cc21c8dbc179c1d116c2dc3